save-orchestrator/src/test/kotlin/org/cqfn/save/orchestrator/controller/heartbeat/HeartbeatControllerTest.kt

```diff
@@ -338,13 +341,16 @@ class HeartbeatControllerTest {
             // todo: should be initialized in @BeforeAll, but it gets called after @DynamicPropertySource
             mockServer = MockWebServer()
             // todo: extract request-based dispatcher to save-cloud-common
```

Comment:
Are you planning to implement this too? There should be a type of dispatcher, that can enqueue responses by path. Now we sometimes face concurrency issues, because order of requests is undetermined, and responses are taken from MockServer sequentially. [This](https://github.com/square/okhttp/tree/master/mockwebserver#dispatcher) is an example of what I'm talkinb about

Reply:
Ok, done it. Now we have a better dispatcher that has some thread-safe queues: on each path you have one queue. Moreover, `checkQueues()` was added in order to check if all queues are empty (have no enqueued responses with non-empty body)